addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/SpiderJob.java

```diff
@@ -236,4 +262,88 @@ public Object getParamMethodObject() {
     public String getParamMethodName() {
         return OPTIONS_METHOD_NAME;
     }
+
+    public static class UrlRequester {
+
+        private HttpSender httpSender;
+        private String requester;
+
+        public UrlRequester(String requester) {
+            this.requester = requester;
+            httpSender =
+                    new HttpSender(
+                            Model.getSingleton().getOptionsParam().getConnectionParam(),
+                            true,
+                            HttpSender.MANUAL_REQUEST_INITIATOR);
+        }
+
+        public void requestUrl(String url, AutomationProgress progress) {
+            // Request the URL
+            try {
+                final HttpMessage msg = new HttpMessage(new URI(url, true));
+                httpSender.sendAndReceive(msg, true);
+
+                if (msg.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+                    progress.error(
+                            Constant.messages.getString(
+                                    "automation.error.spider.url.notok",
+                                    requester,
+                                    url,
+                                    msg.getResponseHeader().getStatusCode()));
+                    return;
+                }
+
+                if (msg.getResponseHeader().isEmpty()) {
```

Comment:
Can this happen? The message it's either sent or an exception happens.

Reply:
Probably copied over from the Quick Scan equivalent ;)